File: Data Structures & Algorithms Mastery/Graphs/Simple Graph/Java/Simple Graph/src/Vertex.java
import java.util.Arrays;
import java.util.LinkedList;

public class Vertex {
    private String data;
    private int key;
    private LinkedList<Vertex> neighbors = new LinkedList<Vertex>();

    public Vertex() {}

    public Vertex(String data) {
        this.data = data;
    }

    public Vertex(String data, int key) {
        this.data = data;
        this.key = key;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getData() {
        return this.data;
    }

    public void setKey(int key) {
        this.key = key;
    }

    public int getKey() {
        return this.key;
    }

    public LinkedList<Vertex> getNeighbors() {
        return this.neighbors;
    }

    @Override
    public String toString() {
        return "\tData: {" + this.data + "} | Key: {" + this.key + "}\n\tNeighbors: " + Arrays.toString(this.neighbors.toArray());
    }
}

File: My CS Projects/NJ Transit Network/nj-transit-network/Vertex.cs
namespace nj_transit_network;

public class Vertex {
    private string data;
    List<Edge> neighbors;

    public Vertex() {
        this.data = "N/A";
        this.neighbors = new List<Edge>();
    }

    public Vertex(string data) {
        this.data = data;
        this.neighbors = new List<Edge>();
    }

    public void SetData(string data) {
        this.data = data;
    }

    public string GetData() {
        return this.data;
    }

    override public string ToString() {
        return this.data;
    }

    public void AddNeighbor(Edge neighbor) {
        this.neighbors.Add(neighbor);
    }

    public void RemoveNeighbor(Edge neighbor) {
        this.neighbors.Remove(neighbor);
    }
}